octopuslite/utils.py

```diff
@@ -155,15 +157,16 @@ def parse_filename(filename: os.PathLike) -> dict:
 
     metadata = {
         "filename": filename,
-        "channel": Channels(int(params.group('channel'))),
-        "time": params.group('time'),
-        "position": params.group('position'),
-        "z": params.group('z'),
+        "channel": Channels(int(params.group("channel"))),
+        "time": params.group("time"),
+        "position": params.group("position"),
+        "z": params.group("z"),
         # "timestamp": os.stat(filename).st_mtime,
     }
 
     return metadata
 
+
 def image_generator(files, crop: Optional[tuple] = None):
```

Comment:
Can you set the return type as a generator (https://docs.python.org/3/library/typing.html#typing.Generator) and add the detail to the docstrings (https://numpydoc.readthedocs.io/en/latest/format.html), e.g.

```
Yields
------
img...
```